pkg/storegateway/series_refs.go

```diff
@@ -712,19 +712,19 @@ func (l *limitingSeriesChunkRefsSetIterator) Err() error {
 }
 
 type loadingSeriesChunkRefsSetIterator struct {
-	ctx                  context.Context
-	postingsSetIterator  *postingsSetsIterator
-	indexr               *bucketIndexReader
-	indexCache           indexcache.IndexCache
-	stats                *safeQueryStats
-	blockID              ulid.ULID
-	shard                *sharding.ShardSelector
-	seriesHasher         seriesHasher
-	skipChunks           bool
-	minTime, maxTime     int64
-	tenantID             string
-	chunkRangesPerSeries int
-	logger               log.Logger
+	ctx                         context.Context
+	postingsSetIterator         *postingsSetsIterator
+	indexr                      *bucketIndexReader
+	indexCache                  indexcache.IndexCache
+	stats                       *safeQueryStats
+	blockID                     ulid.ULID
+	shard                       *sharding.ShardSelector
+	seriesHasher                seriesHasher
+	skipChunks, streamingSeries bool
```

Comment:
we talked about having these as an enum value. Can you do that? Currently there are only 3 possible combinations anyways 

* skipChuks, streamingSeries
* true, false
* false, true
* false, false

maybe something like

```go
type chunkRefsLoadingStrategy int

const (
  dontFilterTime chunkRefsLoadingStrategy = iota // current behaviour when skipChunks=true
  withRanges // loads chunk ranges and filters them by time
  withoutRangesFilteredByTime // doesn't load chunk ranges but still does the filtering
)
```

naming isn't great, but something along those lines

Reply:
Enum is turning out to be a little messy. Everywhere where I have to check for only `skipChunks`, I have to check for 2 enums now. So I will use bit flags, example:
```go
type seriesIteratorStrategy byte

const (
	onlySeries          seriesIteratorStrategy = 0b00000001
	overlappingMintMaxt                        = 0b00000010
)

// Add useful methods on seriesIteratorStrategy
```